conda/core/subdir_data.py

```diff
@@ -905,8 +905,11 @@ def cache_fn_url(url, repodata_fn=REPODATA_FN):
     #    are looking for the cache under keys without this.
     if repodata_fn != REPODATA_FN:
         url += repodata_fn
-    md5 = hashlib.md5(ensure_binary(url)).hexdigest()
-    return '%s.json' % (md5[:8],)
+    try:
+        md5 = hashlib.md5(ensure_binary(url))
+    except ValueError:
+        md5 = hashlib.md5(ensure_binary(url), usedforsecurity=False)
```

Comment:
Since we don't use MD5 for security purposes anyways, I think we should just set `usedforsecurity=False` in all cases and skip the `try`/`except` block. (The code for metadata signature verification and associated package checksumming use SHA-256.)